tests/interpreters/test_onnx_interpreter.py

```diff
@@ -85,3 +97,372 @@ def test_onnx_relu():
     a = ShapedArray([1, 1, 1, 1], [2, 2])
     b = interpreter.run_op(op, (a,))
     assert b == ShapedArray([1, 1, 1, 1], [2, 2])
+
+
+def test_onnx_constant():
+    interpreter = Interpreter(ModuleOp([]))
+    interpreter.register_implementations(OnnxFunctions())
+    interpreter.register_implementations(BuiltinFunctions())
+    op = onnx.Constant(
+        (
+            DenseIntOrFPElementsAttr.create_dense_int(
+                TensorType(i64, [4]), [5, 5, 16, 2]
+            )
+        ),
+        None,
+        None,
+        None,
+        None,
+        None,
+        None,
+        output_type=TensorType(i64, [4]),
+    )
+
+    a = interpreter.run_op(op, ())
+    assert a == ShapedArray([5, 5, 16, 2], [4])
+
+
+def test_onnx_reshape():
+    interpreter = Interpreter(ModuleOp([]))
+    interpreter.register_implementations(OnnxFunctions())
+    op = onnx.Reshape(
+        (TestSSAValue(TensorType(f32, [1, 10]))),
+        (TestSSAValue(TensorType(i64, [2]))),
+        AnyIntegerAttr(0, i64),
+    )
+    a = ShapedArray([1, 2, 3, 4, 5, 6, 7, 8, 9, 10], [1, 10])
+    b = ShapedArray([None], [2])
```

Comment:
The more I think about it, the more I'm sure that the inputs have to determine the size of the output, not the output shape. It's likely that the output shape is dynamic, whereas the input is guaranteed to exist. Like in the tensor reshape, we should test that they don't disagree, and use the input data as the new shape